custom/enikshay/tasks.py

```diff
@@ -0,0 +1,109 @@
+        if (not case.closed and
+           case.type == CASE_TYPE_ADHERENCE and
+           adherence_date > latest_date):
+            latest_date = adherence_date
+            latest_case = adherence_date
+    return latest_case
+
+
+def index_by_adherence_date(adherence_cases):
+    """
+    inde
+    """
+    by_date = defaultdict(list)
+    for case in adherence_cases:
+        adherance_date = parse_date(case.dynamic_case_properties().get('adherence_date'))
+        by_date[adherance_date].append(case)
+    return by_date
+
+def get_adherence_schedule_date_start(episode_case):
+    return parse_datetime(case.dynamic_case_properties().get('adherence_schedule_date_start'))
+
+
+def update_adherence_properties():
+    # edge cases around datetime
+    PURGE_DATE = datetime.datetime.today() - datetime.timedelta(days=60)
+
+    for episode in get_open_episode_cases():
+        adherence_schedule_date_start = get_adherence_schedule_date_start(episode)
+        # Todo: What if adherence_schedule_date_start is None
+        if adherence_schedule_date_start > PURGE_DATE:
+            episode.aggregate_date = adherence_schedule_date_start - 1
+            episode.expected = 0
+            episode.taken = 0
+        else:
+            adherence_case = get_latest_adherence_case_for_episode(episode)
+            # ToDo: what if adherence_case doesn't exist?
+            adherence_date = parse_datetime(adherence_case.dynamic_case_properties().get('adherence_date'))
+            if adherence_date < PURGE_DATE:
+                episode.aggregated_score_date_calculated = adherence_date
+            else:
+                episode.aggregated_score_date_calculated = PURGE_DATE
+
+            # calculate 'aggregated_score_count_taken'
+            adherence_cases = get_adherence_cases_between_dates_from_episode(
+                DOMAIN
+                episode,
+                adherence_schedule_date_start,
+                episode.aggregated_score_date_calculated
+            )
+            adherence_cases_by_date = index_by_adherence_date(adherence_cases)
+            is_dose_taken_by_date = {}
+            for date, cases in adherence_cases_by_date.iteritems():
+                is_dose_taken_by_date[date] = any([
+                    case.adherence_value in DOSE_TAKEN_INDICATORS
+                    for case in cases
+                ])
+            total_taken_count = is_dose_taken_by_date.values().count(True)
+            episode.aggregated_score_count_taken = total_taken_count
+
+            # calculate 'expected' score
+            dose_data = get_doses_data()
+            adherence_schedule_id = episode.adherence_schedule_id or DAILY_SCHEDULE_ID
+            # Todo: what if 'adherence_schedule_id' doesn't exist in fixtures
+            doses_per_week = dose_data.get(adherence_schedule_id)
+            episode.expected = ((aggregated_score_date_calculated - adherence_schedule_date_start) / 7) * doses_per_week
```

Comment:
I think you should call this property `expected_doses_taken` or something more descriptive (also elsewhere)